listener/processors.go

```diff
@@ -185,6 +189,10 @@ func (s *Syncer) processNewBatch(eventName string, abiObject *abi.ABI, vLog *eth
 	if err != nil && gorm.IsRecordNotFoundError(err) {
 		s.Logger.Info("Found a new batch, applying transactions and adding new batch", "index", event.Index.Uint64)
 		_, commitments, err := s.parseAndApplyBatch(vLog.TxHash, event.BatchType)
+		if err == ErrBadBatch {
+			s.Logger.Error("Fraud detected, please challenge batch on-chain", "error", err)
+			panic(err)
```

Comment:
Just highlight that this is the current way we handle the fraud.
Nothing to change here. But we might want to think about the flow to challenge automatically.

Reply:
Yeah there will be a way to autochallenge and plan to do that later, let me highlight that here